src/transports/IframeExecutionEnvironmentTransport.ts

```diff
@@ -0,0 +1,87 @@
+import { Duplex } from 'stream';
+import { Transport } from '@open-rpc/client-js/build/transports/Transport';
+import { IJSONRPCData } from '@open-rpc/client-js/build/Request';
+import { WindowPostMessageStream } from '@metamask/post-message-stream';
+import ObjectMultiplex from '@metamask/object-multiplex';
+import pump from 'pump';
+
+class IframeExecutionEnvironmentTransport extends Transport {
+  private uri: string;
+
+  private stream?: WindowPostMessageStream;
+
+  private commandStream?: Duplex;
+
+  private frame?: Window;
+
+  private postMessageID: string;
+
+  constructor(uri: string) {
+    super();
+    this.uri = uri;
+    this.postMessageID = `post-message-transport-${Math.random()}`;
+  }
+
+  public createWindow(uri: string, timeout = 10000): Promise<Window> {
```

Comment:
maybe this should be longer? something like 60 or 120 seconds? Browsers usually have a longer timeout for http requests, chrome is something like 300 seconds and firefox is 90 seconds